app/controllers/sessions_controller.rb

```diff
@@ -1,33 +1,26 @@
 class SessionsController < ApplicationController
-  def new
-    return unless current_user
-
-    flash[:notice] = 'You are already logged in!'
-    login_redirect(current_user)
-  end
-
-  def login
-    user = User.find_by(email: params[:email])
-    if user && user.authenticate(params[:password])
-      session[:user_id] = user.id
-      flash[:notice] = "Logged in as #{user.name}"
-      redirect_to '/user/dashboard'
-    else
-      flash[:notice] = "Sorry, we don't recognize those credentials."
-      render :new
-    end
+  def create
+    access_token = auth_hash
+    user = User.from_omniauth(access_token)
+    user.google_token = access_token.credentials.token
+    user.name = access_token.info.name.split('@')[0]
+    refresh_token = access_token.credentials.refresh_token
+    user.google_refresh_token = refresh_token if refresh_token.present?
+    user.save
+    session[:user_id] = user.id
+    flash[:notice] = "Logged in as #{user.name}"
+    redirect_to '/user/dashboard'
```

Comment:
This is obviously the biggest change.  This code allows for the sessions to be created with user's Google info.  Nothing other than the user's email is stored on our DB, as Google does all the authentication